components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -360,6 +365,65 @@ void AIChatTabHelper::MakeAPIRequestWithConversationHistoryUpdate(
            {prompt_segment_history, question_part}, nullptr),
        ai_chat::kAIPrompt, " <response>\n"});
 
+  return prompt;
+}
+
+std::string AIChatTabHelper::BuildLlamaPrompt(
+    const mojom::ConversationTurn& turn,
+    bool is_suggested_question) {
+  std::string question_part;
+  // TODO(petemill): Tokenize the summary question so that we
+  // don't have to do this weird substitution.
+  if (turn.text == "Summarize this video") {
+    question_part =
+        l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_SUMMARIZE_VIDEO);
+  } else {
+    question_part = turn.text;
+  }
+
+  std::string prompt = base::StrCat({
+      l10n_util::GetStringUTF8(IDS_AI_CHAT_LLAMA_PROMPT_START),
+      "\n",
+      base::ReplaceStringPlaceholders(
+          l10n_util::GetStringUTF8(
+              is_video_ ? IDS_AI_CHAT_VIDEO_LLAMA_PROMPT_SEGMENT
+                        : IDS_AI_CHAT_ARTICLE_LLAMA_PROMPT_SEGMENT),
+          {article_text_, GetConversationHistoryString(), question_part},
```

Comment:
`BuildLlamaPrompt` doesn't account for conversations that don't have an associated article (i.e. `article_text_` is empty), unlike `BuildClaudePrompt`. Is that intentional?

Reply:
Ah, that makes sense.  I updated so that it works as expected even if article_text_ is empty.